Neos.ContentRepository/Classes/Domain/Utility/NodeContexts.php

```diff
@@ -20,12 +20,12 @@
     /**
      * Determine if the given node path is a context path.
      *
-     * @param string $string path or identifier
+     * @param string $string context path or context identifier
      * @return boolean
      */
     public static function hasContext($string)
     {
-        return (strpos($string, '@') !== false);
+        return (preg_match('/@/u', $string) === 1);
```

Comment:
Fun fact, this was implicitly used everywhere as check for this until I put it into a function. I agree this is naive, but I don't think we really have a good way to determine this.